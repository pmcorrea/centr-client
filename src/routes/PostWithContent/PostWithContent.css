
.PostWithContent {
  background-color: white;
  height: 100%;
  padding: 10px;
  width: 100%;
}

.PostWithContent__content {
  color: #3a3939;
  max-width: 380px;
  padding: 0px 10px 10px;
}
