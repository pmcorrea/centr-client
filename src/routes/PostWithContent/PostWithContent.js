import React, { Component } from "react";
import Post from "../../components/Post/Post";
import "./PostWithContent.css";
import MainContext from "../../contexts/MainContext";
import { findPost } from "../../services/posts-helpers";

export default class PostWithContent extends Component {
  static contextType = MainContext;

  render() {
    const params = this.props.match.params.postId;
    let posts = this.context.posts;
    let post;

    if (params) {
      post = posts ? findPost(posts, parseInt(params)) : {};
    }

    return this.context.posts.length ? (
      <section className="PostWithContent">
        <Post
          id={post.id}
          name={post.post_name}
          modified={post.modified}
          history={this.props.history}
          visibility={post.visibility}
          avatar={post.avatar}
        />
        <div className="PostWithContent__content">
          {post.content.split(/\n \r|\n/).map((para, i) => (
            <p className="PostWithContent__content" key={i}>{para}</p>
          ))}
        </div>
      </section>
    ) : ( "" );
  }
}

PostWithContent.defaultProps = {
  post: {
    content: ""
  }
};
