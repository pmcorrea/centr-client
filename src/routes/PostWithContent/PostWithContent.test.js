import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PostWithContent from './PostWithContent'

describe(`PostWithContent component`, () => {
  const props = {
    post: {
      "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
      "name": "Dogs",
      "modified": "2019-01-03T00:00:00.000Z",
      "content": "Corporis accusamus placeat.\n \rUnde."
    },
    match: {
      params: {
        postId: 0
      }
    }
  }

  it('renders a .PostWithContent by default', () => {
    const wrapper = shallow(<PostWithContent {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Post with post prop', () => {
    const post = shallow(<PostWithContent {...props} />)
      .find('Post')
    expect(toJson(post)).toMatchSnapshot()
  })

  it(`splits the content by \\n or \\n\\r, with a p foreach`, () => {
    [{
      post: { "content": "Content with n r.\n \rafter n r." },
      match: {params:{postId:0}}
    }, {
      post: { "content": "Content with n.\nafter." },
      match: {params:{postId:0}}
    }].forEach(props => {
      const content = shallow(<PostWithContent {...props} />)
        .find('PostWithContent__content')
      expect(toJson(content)).toMatchSnapshot()
    })
  })
})
