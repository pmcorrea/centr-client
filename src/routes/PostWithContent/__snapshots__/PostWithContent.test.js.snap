// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostWithContent component renders a .PostWithContent by default 1`] = `""`;

exports[`PostWithContent component renders a Post with post prop 1`] = `null`;

exports[`PostWithContent component splits the content by \\n or \\n\\r, with a p foreach 1`] = `null`;

exports[`PostWithContent component splits the content by \\n or \\n\\r, with a p foreach 2`] = `null`;
